File: go-sdk/pkg/models/operations/getfilesidtrash.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFilesIDTrashPathParams struct {
	FileID string `pathParam:"style=simple,explode=false,name=file_id"`
}

type GetFilesIDTrashQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type GetFilesIDTrashRequest struct {
	Retries     *utils.RetryConfig
	PathParams  GetFilesIDTrashPathParams
	QueryParams GetFilesIDTrashQueryParams
}

type GetFilesIDTrashResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
	TrashFile   *shared.TrashFile
}

File: ts-sdk/src/sdk/models/operations/getfilesidtrash.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetFilesIdTrashPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;
}


export class GetFilesIdTrashQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}


export class GetFilesIdTrashRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetFilesIdTrashPathParams;

  @SpeakeasyMetadata()
  queryParams: GetFilesIdTrashQueryParams;
}


export class GetFilesIdTrashResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  trashFile?: shared.TrashFile;
}
